reverse_proxy.go

```diff
@@ -302,18 +302,23 @@ type ReverseProxy struct {
 	ErrorHandler ErrorHandler
 }
 
-func defaultTransport() *http.Transport {
-	// allocate a new one every time for now, to avoid modifying a
-	// global variable for each request's needs (e.g. timeouts).
+func defaultTransport(dialerTimeout int) *http.Transport {
+	timeout := 30
+	if dialerTimeout > 0 {
+		log.Debug("Setting timeout for outbound request to: ", dialerTimeout)
+		timeout = dialerTimeout
+	}
+
 	return &http.Transport{
-		DialContext: (&net.Dialer{
-			Timeout:   30 * time.Second,
+		DialContext: dnsCacheManager.WrapDialer(&net.Dialer{
```

Comment:
seems like in situation when we know for sure that DNS cache is disabled (in config) we don't want to wrap Dialer as it introduces extra closure and some logic inside dnsCacheManager to be run

Reply:
I wasn't able to find those cases in code (when cache will be re-enabled). Looks like we enable DNS cache at gateway process start up time and it comes from tyk.conf which gets read again only when we start (or restart) process.

We've spent a while recently on performance optimization. This wrapper is adding extra ops and allocations every time we connect/reconnect to upstream and I think we can easily eliminate it and use standard `net.Dialer` when DNS cache is disabled.

Reply:
So are you suggesting to remove this extension point(by this I mean the ability to detect cache on/off in runtime) in order to make the code performant?

Reply:
I do not care that much about performance here. What I do care about, is that when we wrap dialer like this, we made all our users use this new code (which for sure has bug somewhere, like with all new code). So if we wrap dialer only when DNS turned on, this potential bugs will happen only for users who enabled DNS caching.